rma/models/stock_move.py

```diff
@@ -88,6 +89,7 @@ def _update_reserved_quantity(
             not lot_id
             and self.rma_line_id.lot_id
             and self.location_id.usage == "internal"
+            and self.rma_line_id.operation_id.out_force_same_lot
```

Comment:
I do see this PR a bit late but, are you sure this part is still necessary ?
Indeed, I see this PR now fill the `restrict_lot_id` in the procurement data, and then the module `stock_restrict_lot` should manage the reservations properly.